geode-assembly/geode-assembly-test/src/main/java/org/apache/geode/test/junit/rules/GeodeAdminRestClient.java

```diff
@@ -0,0 +1,88 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more contributor license
+ * agreements. See the NOTICE file distributed with this work for additional information regarding
+ * copyright ownership. The ASF licenses this file to You under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance with the License. You may obtain a
+ * copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package org.apache.geode.test.junit.rules;
+
+import java.nio.charset.StandardCharsets;
+import java.security.SecureRandom;
+
+import javax.net.ssl.KeyManager;
+import javax.net.ssl.SSLContext;
+import javax.net.ssl.TrustManager;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.apache.http.HttpHost;
+import org.apache.http.HttpResponse;
+import org.apache.http.client.methods.HttpPost;
+import org.apache.http.client.methods.HttpRequestBase;
+import org.apache.http.client.protocol.HttpClientContext;
+import org.apache.http.conn.ssl.NoopHostnameVerifier;
+import org.apache.http.entity.StringEntity;
+import org.apache.http.impl.client.HttpClientBuilder;
+import org.apache.http.impl.client.HttpClients;
+
+import org.apache.geode.management.internal.security.ResourceConstants;
+
+public class GeodeAdminRestClient {
+  private static String CONTEXT = "/geode-mgmt/v2";
+
+  private String context;
+  private HttpHost host;
+  private boolean useSsl;
+
+  public GeodeAdminRestClient(String bindAddress, int restPort, boolean useSsl) {
+    this.context = CONTEXT;
```

Comment:
Might something like `basePath` be a more descriptive name?

Reply:
`context` means a webapp's name in here. Essentially we have two webapps hosted in our jetty server, `geode` and `geode-mgmt`, one for dev rest, one for admin rest. The name here is the `context`.